addons/account/models/account_move_line.py

```diff
@@ -2505,9 +2505,15 @@ def _prepare_exchange_difference_move_vals(self, amounts_list, company=None, exc
         expense_exchange_account = company.expense_currency_exchange_account_id
         income_exchange_account = company.income_currency_exchange_account_id
 
+        temp_exchange_move = self.env['account.move'].new({'journal_id': journal.id})
+        accounting_exchange_date = temp_exchange_move._get_accounting_date(
+            exchange_date or fields.Date.today(),
```

Comment:
@smetl Does it really make sense to have `exchange_date` as an optional parameter to this function ?
I was tempted to make it mandatory and thus be able to remove this 'or' condition here.
I checked and every usage of the function specifies the `exchange_date` at the moment.

Reply:
@clbr-odoo I know there is a reason for that but I don't remember :see_no_evil:

Reply:
So we keep it like that ? :rofl: